kernel/init.c

```diff
@@ -424,6 +496,8 @@ void z_early_boot_rand_get(uint8_t *buf, size_t length)
 __boot_func
 FUNC_NORETURN void z_cstart(void)
 {
+	z_sys_init_run_level(INIT_LEVEL_ARCH);
```

Comment:
I do not think the `ARCH` level is supposed to do that based on the PR that introduced it; but, I agree it is quite vague in its current form.

Reply:
The problem is that PR introducing such level was merged without proper reviewing, no docs, etc. So, who knows what "ARCH" init level means now.